### PR TITLE
cli: update init for image builds

### LINKED ISSUE #207: update ci for projects
we should update the CI to work with the image build setup (see https://github.com/SwissDataScienceCenter/renku/pull/260). This should include:

`.gitlab-ci.yml`:

```
stages:
  - build

image_build:
  stage: build
  image: python:3.6
  before_script:
    - curl -sSL https://get.docker.com/ | sh
    - docker login -u gitlab-ci-token -p $CI_JOB_TOKEN http://$CI_REGISTRY
  script:
    - CI_COMMIT_SHA_7=$(echo $CI_COMMIT_SHA | cut -c1-7)
    - docker build --tag $CI_REGISTRY_IMAGE:$CI_COMMIT_SHA_7 .
    - docker push $CI_REGISTRY_IMAGE:$CI_COMMIT_SHA_7
  tags:
    - image-build

dot:
  stage: build
  image: renku/renku-python:latest
  script:
    - python -c "import renku, networkx; graph = renku.cli._graph.Graph(renku.LocalClient()); graph.build_status(); networkx.drawing.nx_pydot.write_dot(graph.G, 'graph.dot')"
  artifacts:
    paths:
      - graph.dot
  environment:
    name: dot/$CI_COMMIT_REF_NAME
    url: $CI_PROJECT_URL/-/jobs/artifacts/$CI_COMMIT_REF_NAME/raw/graph.dot?job=$CI_JOB_NAME
```

`Dockerfile`:

```
FROM renku/singleuser:latest

COPY requirements.txt /tmp/requirements.txt

COPY src /code/src

WORKDIR /code

USER 0

RUN pip install -r /tmp/requirements.txt

USER 1000
```

a blank `requirements.txt`

### PR DESCRIPTION
**Pull request by rokroskar**
has been tested against https://github.com/SwissDataScienceCenter/renku/pull/260

closes #207

### CODE REVIEW

File: renku/cli/.gitlab-ci.yml [hunk 1]
```diff
@@ -1,8 +1,20 @@
-# renku: v1.dev20180404
+# renku: v1.dev20181906
```
Comment (jirikuncar) [inline renku/cli/.gitlab-ci.yml]: 20180619 - `YYYYMMDD` so it's sortable

File: renku/cli/runner.py [hunk 2]
```diff
@@ -39,10 +39,12 @@ def runner():
 @runner.command()
 @pass_local_client
 def template(client):
-    """Generate template for CI."""
-    with open(client.path / _GITLAB_CI, 'wb') as dest:
-        with pkg_resources.resource_stream(__name__, _GITLAB_CI) as tpl:
-            dest.write(tpl.read())
+    """Render templated configuration files."""
+    # create the templated files
+    for tpl_file in [_GITLAB_CI, 'Dockerfile', 'requirements.txt']:
```
Comment (jirikuncar) [inline renku/cli/runner.py]: can you define the files above as it is done for `_GITLAB_CI`?

```python
CI_TEMPLATES = (_GITLAB_CI, ...)
```

File: renku/cli/.gitlab-ci.yml [hunk 3]
```diff
@@ -15,14 +27,3 @@ dot:
   environment:
     name: dot/$CI_COMMIT_REF_NAME
     url: $CI_PROJECT_URL/-/jobs/artifacts/$CI_COMMIT_REF_NAME/raw/graph.dot?job=$CI_JOB_NAME
-
-review_notebook:
```
Comment (jirikuncar) [inline renku/cli/.gitlab-ci.yml]: I would leave it here so one can access the URL via GitLab.

Comment (rokroskar) [reply renku/cli/.gitlab-ci.yml]: It's a simple change to put it back in if it's needed in the future -- right now it's not being used anywhere so it shouldn't be included.

**Pull request merged**